lib/matplotlib/backends/backend_agg.py

```diff
@@ -357,7 +357,10 @@ def restore_region(self, region, bbox=None, xy=None):
             else:
                 ox, oy = xy
 
-            self._renderer.restore_region(region, x1, y1, x2, y2, ox, oy)
+            # The incoming data is float, but the _renderer type-checking wants
+            # to see integers.
+            self._renderer.restore_region(region, int(x1), int(y1),
+                                          int(x2), int(y2), int(ox), int(oy))
```

Comment:
To be pedantic, not all of the incoming data is coming in as floats. For example, `region.get_extents()` will return integers, which is probably why we didn't notice this bug as I think that is the typical code-path. I haven't checked what `bbox.extents` would return, but certainly input-sanitation should be done anyway due to the possibility of the user passing in a tuple of floats as a bbox.

How did you trigger the error?